ntbs-service/Models/Notification.cs

```diff
@@ -1,18 +1,127 @@
+                return CreateTimePeriodsString(SocialRiskFactors.RiskFactorDrugs);
+            }
+        }
+
+        public int? DaysFromOnsetToTreatment {
+            get {
+                return CalculateDaysBetweenNullableDates(ClinicalDetails.TreatmentStartDate, ClinicalDetails.SymptomStartDate);
+            }
+        }
+
+        public int? DaysFromOnsetToPresentation {
+            get {
+                return CalculateDaysBetweenNullableDates(ClinicalDetails.PresentationDate, ClinicalDetails.SymptomStartDate);
+            }
+        }
+
+        public int? DaysFromPresentationToDiagnosis {
+            get {
+                return CalculateDaysBetweenNullableDates(ClinicalDetails.DiagnosisDate, ClinicalDetails.PresentationDate);
+            }
+        }
+
+        public int? DaysFromDiagnosisToTreatment {
+            get {
+                return CalculateDaysBetweenNullableDates(ClinicalDetails.TreatmentStartDate, ClinicalDetails.DiagnosisDate);
+            }
+        }
+
+        public string BCGVaccinationStateAndYear {
+            get {
+                return FormatVaccintationStateAndYear(ClinicalDetails.BCGVaccinationState, ClinicalDetails.BCGVaccinationYear);
+            }
+        }
+
+        public string FormatVaccintationStateAndYear(Status vaccinationState, int? vaccinationYear) {
+            return vaccinationState.ToString() + (vaccinationYear != null ? " - " + vaccinationYear : "");
+        }
+
+        public int? CalculateDaysBetweenNullableDates(DateTime? date1, DateTime? date2) {
+            return (date1?.Date - date2?.Date)?.Days;
+        }
+
+        public string CreateSitesOfDiseaseString() {
+            var siteNames = NotificationSites.Select(ns => ns.Site).Select(s => s.Description);
+            return String.Join(", ", siteNames); 
+        }
+
+        public string CreatePostcodeString() {
+            var postcodeNoWhiteSpace = PatientDetails.Postcode.Replace(" ", string.Empty);
+            string FormattedPostcode = postcodeNoWhiteSpace.Insert(postcodeNoWhiteSpace.Length - 3, " ");
+            return FormattedPostcode;
+        }
+
+        public string CreateFormattedNhsNumberString() {
+            return PatientDetails.NhsNumber.ToString().Substring(0, 3) + " " + PatientDetails.NhsNumber.ToString().Substring(3, 3) 
+                + " " + PatientDetails.NhsNumber.ToString().Substring(6, 4);
+        }
+
+        public string CreateTimePeriodsString(RiskFactorBase riskFactor) {
+            string timeString = "";
+            if(riskFactor.IsCurrent) {
+                timeString = "current";
+            }
+            if(riskFactor.InPastFiveYears) {
+                timeString = String.Join(", ", new string[] {timeString, "less than 5 years ago"});
+            }
+            if(riskFactor.MoreThanFiveYearsAgo) {
+                timeString = String.Join(", ", new string[] {timeString, "more than 5 years ago"});
+            }
+            return timeString;
```

Comment:
Sorry I don't think my comment about `String.Join` was super clear before. One of th enice things about it is it can take an array, so we can build up the array first and then concatenate it in one go. So something along the lines of:
```
            var timeStrings = new List<strings>();
            if(riskFactor.IsCurrent) {
                timeStrings.Add("current");
            }
            if(riskFactor.InPastFiveYears) {
                timeStrings.Add("less than 5 years ago");
            }
            if(riskFactor.MoreThanFiveYearsAgo) {
                timeStrings.Add("more than 5 years ago");
            }
           String.Join(", ", timeStrings);
```